DOC: Update start/end parameter for BusinessHour and CustomBusinessHour

gandhis1 opened the pull request:
These are all valid usages. Examples from pandas-stubs here:

https://github.com/pandas-dev/pandas-stubs/blob/main/tests/test_timefuncs.py#L601-L639

gandhis1 committed:
Update start/end parameter documentation for BusinessHour and CustomBusinessHour to mention allowed input types of time and list of str/time

mroeschke commented on pandas/_libs/tslibs/offsets.pyx:
```diff
@@ -1613,9 +1613,9 @@ cdef class BusinessHour(BusinessMixin):
     weekmask : str, Default 'Mon Tue Wed Thu Fri'
         Weekmask of valid business days, passed to ``numpy.busdaycalendar``.
-    start : str, default "09:00"
+    start : str, time, or list of str/time, default "09:00"
```
Could you add to the `Examples` section showing this?

gandhis1 replied:
@mroeschke Done

gandhis1 committed:
Add additional examples to pd.offsets.BusinessHour and CustomBusinessHour

This covers usage of datetime.time start/end values as well as multiple start and end values

gandhis1 commented on pandas/_libs/tslibs/offsets.pyx:
```diff
@@ -3619,16 +3626,23 @@ cdef class CustomBusinessHour(BusinessHour):
+    >>> ts + pd.offsets.CustomBusinessHour(end=dt_time(19, 0))
+    Timestamp('2022-08-05 17:00:00')
+    >>> ts + pd.offsets.CustomBusinessHour(start=[dt_time(9, 0), "20:00"],
+    ...                                    end=["17:00", dt_time(22, 0)])
```
The docstrings examples apparently get compiled. This is failing. I tried `time`, but the error seemed to indicate it uses the same imports as the module itself - so I updated to using `dt_time`, which still didn't work. Based on the following import:

```
from cpython.datetime cimport (
    PyDate_Check,
    PyDateTime_Check,
    PyDelta_Check,
    date,
    datetime,
    import_datetime,
    time as dt_time,
    timedelta,
)
```

I would have expected `dt_time` to work. What's the right syntax here?

mroeschke commented on pandas/_libs/tslibs/offsets.pyx:
```diff
@@ -1615,16 +1615,23 @@ cdef class BusinessHour(BusinessMixin):
     Timestamp('2022-08-08 09:00:00')
+    >>> ts + pd.offsets.BusinessHour(start="11:00")
+    Timestamp('2022-08-08 11:00:00')
+    >>> ts + pd.offsets.BusinessHour(end=dt_time(19, 0))
```
You'll need to import `dt_time` before the first usage.

`>>> from datetime import time as dt_time`

mroeschke commented:
Thanks @gandhis1
